### DISCUSSION TITLE
HammerDB 4.7 + SQL Server 2022 + TPROC-H : Data loading issue

### DISCUSSION THREAD

**Discussion by mmareddy943** [TPC-Council/HammerDB#532] [Q&A] [answered]
Hi Team,

I am getting an issue while generating the data using HammerDB 4.7 and SF 300/1000. Here, I am using the HDB GUI method. i have tried multiple times but i got the same error and stuck in the final stage. I have used 51 users. in middle, one user failed for the below reason and the remaining workers are completed. But the virtual user1 is waiting for virtual user 25 to complete. 

ERROR:
Error in Virtual User 25: Error: [Microsoft][ODBC Driver 17 for SQL Server]TCP Provider: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

[Microsoft][ODBC Driver 17 for SQL Server]Communication link failure

But i don't know how to proceed with the next steps. Can you help me with this?

**Comment by sm-shaw** [comment, marked as answer]
The error  [Microsoft][ODBC Driver 17 for SQL Server]Communication link failure is a SQL Server error rather than a HammerDB one. The best place to troubleshoot is in the application logs to see if there is any indication to why this would be the case. It can be a network issue or the system is overloaded doing the build. You don't mention the configuration of the system you are loading. Sometimes using fewer virtual users can be faster than more, depending on the resources available. 

If there are known network issues such as loading a cloud database from a remote location, it would be better to use the datagen feature to create flat files and then do a bulk load https://www.hammerdb.com/docs/ch13.html. This takes more time and configuration than doing an insert based direct load without staging files first, however with a direct load you are generating redo for all of the transactions so there is much greater overhead in keeping the database in synch and therefore a datagen based load might be quicker in the long run.

**Reply by mmareddy943** [reply]
I have checked the logs and loading percentage also.. both are fine.. My resource utilization is not even 40%.. Even with a small number also i am seeing this issue. Especially lineitem loading time this issue has happened. Here i am using VMs on on-premises only. Yeah we can generate the staging files and then use bulk-insert command to load the data also. Thanks for your reply and inputs. I will go with datagen method this time instead of direct loading.
